core/src/com/unciv/logic/map/TileMap.kt

```diff
@@ -152,7 +152,7 @@ class TileMap {
 
     fun getViewableTiles(position: Vector2, sightDistance: Int, ignoreCurrentTileHeight:Boolean=false): MutableList<TileInfo> {
         val viewableTiles = getTilesInDistance(position, 1).toMutableList()
-        val currentTileHeight = if(ignoreCurrentTileHeight) 0 else get(position).getHeight()
+        val currentTileHeight = if (ignoreCurrentTileHeight) Int.MAX_VALUE else get(position).getHeight()
```

Comment:
if ignore -> this is the highest tile in game

Reply:
I see the `ignoreCurrentTileHeight` is set to `true` only for water units and cities. In the first case I think it makes sense to have height to 0, probably would be better to change `getHeight` though?
Cities instead are suppose to be able to see ALL tiles in a range of 2.

As @naivekook points out, I think `ignoreCurrentTileHeight` should give the unit "super sight". My proposal is to switch to `getTilesInDistance` in case `ignoreCurrentTileHeight` is `true` and modify `TileInfo.getHeight` to work with water tiles.

Reply:
Agree that ignoreCurrentTilHeight should give us getTilesInDistance because it's very readable and less CPU intensive

Reply:
accepted